### PR TITLE
ref(ts): Convert Projects Dashboard to typescript

### PR DESCRIPTION
**Pull request by scttcper**
Removes unused newIssues component
Swap out deprecated reflexbox for css grid
Fix "Resources" on mobile

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectsDashboard/index.tsx [hunk 1]
```diff
@@ -53,7 +64,7 @@ class Dashboard extends React.Component {
     const filteredTeams = teams.filter(team => team.projects.length);
     filteredTeams.sort((team1, team2) => team1.slug.localeCompare(team2.slug));
 
-    const projects = uniq(flatten(teams.map(teamObj => teamObj.projects)), 'id');
+    const projects = uniqBy(flatten(teams.map(teamObj => teamObj.projects)), 'id');
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/projectsDashboard/index.tsx:67]: I'm not sure how this was working, maybe it wasn't.

File: src/sentry/static/sentry/app/views/projectsDashboard/projectCard.tsx [hunk 2]
```diff
@@ -14,26 +14,33 @@ import ProjectsStatsStore from 'app/stores/projectsStatsStore';
...
   static propTypes = {
     organization: SentryTypes.Organization.isRequired,
     project: SentryTypes.Project.isRequired,
     hasProjectAccess: PropTypes.bool,
   };
 
   componentDidMount() {
-    const {organization, project} = this.props;
-
-    this.api = new Client();
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/projectsDashboard/projectCard.tsx:32]: using withApi for this

File: tests/js/spec/views/projectsDashboard/projectCard.spec.jsx [hunk 3]
```diff
@@ -36,23 +36,20 @@ describe('ProjectCard', function() {
   });
 
   it('renders latest 2 deploys', function() {
-    const latestDeploys = [
-      {
-        environment: 'beta',
+    const latestDeploys = {
```
Comment (scttcper) [inline tests/js/spec/views/projectsDashboard/projectCard.spec.jsx:39]: Updated latestDeploys format to match api response

File: src/sentry/static/sentry/app/views/projectsDashboard/resources.tsx [hunk 4]
```diff
@@ -11,9 +12,18 @@ import releasesImg from '../../../images/spot/releases.svg';
...
+type Props = {
+  organization: Organization;
+};
+
+class Resources extends React.Component<Props> {
   componentDidMount() {
-    analytics('orgdash.resources_shown');
+    const {organization} = this.props;
+    trackAnalyticsEvent({
+      eventKey: 'orgdash.resources_shown',
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/projectsDashboard/resources.tsx:23]: Is the eventKey the same as `analytics('orgdash.resources_shown');`, does it matter?

File: src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+      <Version
+        version={deploy.version}
+        projectId={project.id}
+        tooltipRawVersion
+        truncate
+      />
+    </StyledTextOverflow>
+
+    <DeployTimeWrapper>
+      <TimeSince date={deploy.dateFinished} />
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx:74]: using existing TimeSince component instead

File: src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+          deploy={deploy}
+          project={project}
+        />
+      ))}
+    </DeployContainer>
+  );
+};
+
+Deploys.propTypes = {
+  project: SentryTypes.Project.isRequired,
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx:51]: removed unused organization prop

File: src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+};
+
+export default Deploys;
+
+type DeployProps = Omit<Props, 'organization'> & {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx:57]: `Props` doesn't have `organization` anymore, do you still need this Omit?

File: src/sentry/static/sentry/app/views/projectsDashboard/projectCard.tsx [hunk 8]
```diff
@@ -42,7 +49,7 @@ class ProjectCard extends React.Component {
     const {id, firstEvent, stats, slug} = project;
 
     return (
-      <ProjectCardWrapper data-test-id={slug} width={['100%', '50%', '33%', '25%']}>
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/projectsDashboard/projectCard.tsx:45]: Was this width prop unused? It looks reflexbox related.

File: src/sentry/static/sentry/app/views/projectsDashboard/resources.tsx (hunk 4 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/projectsDashboard/resources.tsx:23]: They look the same to me :+1:

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/projectsDashboard/projectCard.tsx (hunk 8 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/projectsDashboard/projectCard.tsx:45]: Yes, this was reflexbox. Replaced with the css grid layout

File: src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx (hunk 7 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx]: I didn't know you could omit things that didn't exist, interesting!

File: src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  color: ${p => p.theme.gray500};
+  font-size: 12px;
+
+  &:not(:last-of-type) {
+    margin-top: ${space(1)};
+  }
+`;
+
+const Environment = styled(TextOverflow)`
+  font-size: 11px;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx:110]:
```suggestion
  font-size: ${p => p.theme.fontSizeExtraSmall};
```

File: src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+      </Button>
+    </Background>
+  </DeployContainer>
+);
+
+const DeployRow = styled('div')`
+  display: flex;
+  justify-content: space-between;
+  color: ${p => p.theme.gray500};
+  font-size: 12px;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectsDashboard/deploys.tsx:102]:
```suggestion
  font-size: ${p => p.theme.fontSizeSmall};
```

File: src/sentry/static/sentry/app/views/projectsDashboard/teamSection.tsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  display: grid;
+  grid-template-columns: minmax(100px, 1fr);
+  grid-gap: ${space(3)};
+
+  @media (min-width: ${p => p.theme.breakpoints[0]}) {
+    grid-template-columns: repeat(2, minmax(100px, 1fr));
+  }
+
+  @media (min-width: ${p => p.theme.breakpoints[2]}) {
+    grid-template-columns: repeat(3, minmax(100px, 1fr));
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectsDashboard/teamSection.tsx:65]: wouldn't it be better if you use `repeat(auto-fill `here?

File: src/sentry/static/sentry/app/views/projectsDashboard/teamMembers.tsx [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+  teamId: string;
+  orgId: string;
+};
+
+type State = AsyncComponent['state'] & {
+  members?: Member[];
+};
+
+class TeamMembers extends AsyncComponent<Props, State> {
+  static propTypes = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectsDashboard/teamMembers.tsx:18]: do you still need the propTypes here?

File: src/sentry/static/sentry/app/views/projectsDashboard/teamSection.tsx (hunk 11 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/projectsDashboard/teamSection.tsx:65]: used auto-fit instead!

File: src/sentry/static/sentry/app/views/projectsDashboard/teamMembers.tsx (hunk 12 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/projectsDashboard/teamMembers.tsx:18]: I don't think we've been removing them yet

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/projectsDashboard/teamMembers.tsx:18]: we are removing them if they are no longer being used 😉

**Pull request merged**